scripts/release-checks/dependency-ranges/index.mts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import semver from 'semver';
+import { Log, bold, ReleasePrecheckError } from '@angular/ng-dev';
+import { checkPeerDependencies } from './peer-deps-check.mjs';
+import { checkSchematicsAngularLatestVersion } from './latest-versions-check.mjs';
+import { PackageMap } from '../../../lib/packages.js';
+
+/** Environment variable that can be used to skip this pre-check. */
+const skipEnvVar = 'SKIP_DEPENDENCY_RANGE_PRECHECK';
+
+/**
+ * Ensures that dependency ranges are properly updated before publishing
+ * a new version. This check includes:
+ *
+ *   - checking of `latest-versions.ts` of `@schematics/angular`.
+ *   - checking of peer dependencies in `@angular-devkit/build-angular`.
+ *
+ * @throws {ReleasePrecheckError} If validation fails.
+ */
+export async function assertValidDependencyRanges(
+  newVersion: semver.SemVer,
+  allPackages: PackageMap,
+) {
+  if (process.env[skipEnvVar] === '1') {
```

Comment:
Nit: I could see someone trying `SKIP_DEPENDENCY_RANGE_PRECHECK="true"` and being confused when that doesn't work. I wonder if we should skip if the variable is truthy or at least throw an error that the only valid value is `1` or unset?

Reply:
Honestly, I don't know it's worth doing that. the error is pretty clear about what needs to be used, and it's also only the experienced caretakers running the release process.